src/validation.cpp

```diff
@@ -969,12 +1056,98 @@ bool MemPoolAccept::PackageMempoolChecks(const std::vector<CTransactionRef>& txn
+        direct_conflict_iters.merge(ws.m_iters_conflicting);
+    }
+
+    // Use the child as the transaction for attributing errors to.
+    const auto hash = workspaces[workspaces.size() - 1].m_ptx->GetHash();
+    const CFeeRate package_feerate(m_total_modified_fees, m_total_vsize);
+
+    // Calculate all conflicting entries and enforce Rules 2 and 5.
+    for (Workspace& ws : workspaces) {
+        // The aggregated set of conflicts cannot exceed 100.
+        if (const auto err_string{GetEntriesForConflicts(*ws.m_ptx, m_pool, direct_conflict_iters,
+                                                         m_all_conflicts)}) {
+            return package_state.Invalid(PackageValidationResult::PCKG_POLICY,
+                                         "package RBF failed: too many potential replacements", *err_string);
+        }
+    }
+
+    // Make sure none of the package transactions depend on a mempool transaction that is going to be replaced.
+    std::set<uint256> all_conflicting_txids;
+    std::transform(m_all_conflicts.cbegin(), m_all_conflicts.cend(),
+                   std::inserter(all_conflicting_txids, all_conflicting_txids.end()),
+                   [](const auto& entry) { return entry->GetTx().GetHash(); });
+    if (const auto err_string{EntriesAndTxidsDisjoint(m_collective_ancestors, all_conflicting_txids, hash)}) {
+        // Note that we handle this differently in individual transaction validation (a transaction
+        // that conflicts with its own dependency is inconsistent, but this could just be
+        // conflicting transactions in a package).
+        return package_state.Invalid(PackageValidationResult::PCKG_POLICY,
+                                     "package RBF failed: package conflicts with dependency", *err_string);
+    }
+
+    // CheckMinerScores is very conservative and should not be used for individual transactions.
+    // For example, the mempool contains a large, low-feerate transaction A (99,000vB, 1sat/vB feerate) is
+    // Transaction A has a small, high-feerate child B (1,000vB, 101sat/vB). The user wants to
+    // further bump A+B by replacing B with an even higher feerate transaction, B'. If
+    // CheckMinerScores is enforced, then B' needs an ancestor score higher than the individual
+    // feerate of its directly conflicting transaction, B, which is 101sat/vB, This is extremely
+    // expensive since the ancestor feerate includes A (101sat/vB * 99,000vB).
+    // On the other hand, CheckMinerScores is fine if A+B is to be replaced by A'+B' (where A' is a
+    // transaction that conflicts with A) because the directly conflicting transaction A, has a low
+    // individual feerate.
+    // As such, it's also important to ensure that we don't apply CheckMinerScores() to individual
+    // transactions that were submitted as a package (e.g. if it's the only transaction left after
+    // deduplication in AcceptPackage()).
+    Assume(txns.size() > 1);
```

Comment:
errr, the same code path hits here :)

Reply:
Oops I lost a commit on the floor, thanks.
I think this assumption should hold true now? `AcceptPackage` should now be calling `AcceptSingleTransaction`, not `AcceptMultipleTransactions` when there's only 1 tx left. Which means there should always be more than 1 tx when we get here.